data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -100,9 +106,13 @@ public StreamWorker(final RecordBufferWriter recordBufferWriter,
         this.successItemsCounter = pluginMetrics.counter(SUCCESS_ITEM_COUNTER_NAME);
         this.failureItemsCounter = pluginMetrics.counter(FAILURE_ITEM_COUNTER_NAME);
         this.bytesReceivedSummary = pluginMetrics.summary(BYTES_RECEIVED);
+        this.executorService = Executors.newSingleThreadExecutor(BackgroundThreadFactory.defaultExecutorThreadFactory("mongodb-stream-checkpoint"));
         if (sourceConfig.isAcknowledgmentsEnabled()) {
```

Comment:
should this be `!sourceConfig.isAcknowledgmentsEnabled()`?

Reply:
when `sourceConfig.isAcknowledgmentsEnabled()` it uses `streamAcknowledgementManager`, otherwise checkpoint is done in the private method `checkpointStream`

Reply:
I see. I might have confused the checkpoint concept with that in e2e ack.